pyvista/core/utilities/points.py

```diff
@@ -459,3 +465,112 @@ def vector_poly_data(orig, vec):
     pdata.GetPointData().SetActiveScalars(name)
 
     return pyvista.PolyData(pdata)
+
+
+def principal_axes(points: MatrixLike[float]):
+    """Compute the principal axes of a set of points.
+
+    Principal axes are orthonormal vectors that best fit a set of points. The axes
+    are also known as the principal components of the points in Principal Component
+    Analysis (PCA), or the right singular vectors from the Singular Value Decomposition
+    (SVD).
+
+    The axes are computed as the eigenvectors of the covariance matrix from the
+    mean-centered points, and are processed to ensure that they form a right-handed
+    coordinate frame.
+
+    The axes explain the total variance of the points. The first axis explains the
+    largest percentage of variance, followed by the second axis, followed again by
+    the third axis which explains the smallest percentage of variance. The axes may
+    be used to build an oriented bounding box of the points or to transform the points
+    so the principal axes of the points are aligned with the XYZ axes.
+
+    .. note::
+        The computed axes are not unique, and the sign of each axis direction can be
+        arbitrarily changed.
+
+    .. versionadded:: 0.44.0
+
+    Parameters
+    ----------
+    points : MatrixLike[float]
+        Nx3 array of points. At least two points are required.
+
+    Returns
+    -------
+    numpy.ndarray
+        3x3 orthonormal array with the principal axes as row vectors.
```

Comment:
Do you think it would also be interesting to (optionally??) return the sizes? We already have them.

Reply:
This OBB implementation with VTK is a refactored version from what was originally implemented in #4947, which used numpy and SVD originally (and the SVD uses LAPACK under the hood) 
In that implementation the SVD values were also extracted. In my testing with this VTK version I was finding there wasn't a good correlation between the SVD values from numpy and the eigen values (sizes) from this VTK version (im not an expert on PCA and SVD but I thought they would correlate by a sqrt factor).

I looked at the VTK source code and it uses vtkMathJacobi for the computation, which returns the vectors and size. But those are returned as column vectors. So I'm just not sure if the sizes actually correlate correctly with the vectors. Like, what do they mean? Are they the eigen values? So I don't have confidence about what the `size` represents

Reply:
Good point. The documentation says
> Also return a sorted list of relative "sizes" of axes for comparison purposes. 

which doesn't spark confidence.

Reply:
The implementation has changed to use numpy, and so the `sizes` are now optionally returned as suggested in https://github.com/pyvista/pyvista/pull/6346/commits/566c6ba1a403a6f7ef8d33a20ea66371b04b21ae